airflow/triggers/external_task.py

```diff
@@ -80,7 +80,7 @@ async def run(self) -> typing.AsyncIterator["TriggerEvent"]:
         while True:
             num_tasks = await self.count_tasks()
             if num_tasks == len(self.execution_dates):
-                yield TriggerEvent(True)
+                yield TriggerEvent(self.serialize())
```

Comment:
I think in the operator `execute_complete` method we are only using `execution_date` so shall we yield only that value 

```yield TriggerEvent({"execution_date": self.execution_dates[0]})```

then in `execute_complete`  we can do
```
parsed_execution_date = event["execution_date"]
```
wdyt?

Reply:
I think yielding more from the trigger is preferable to restricting the interface for this one user case. It provides for future use cases without needing a breaking change.

Reply:
I agree with @dylanbstorey, I prefer to you the serialized object too.